src/System Application/App/Email/src/EmailInstaller.Codeunit.al

```diff
@@ -46,21 +48,73 @@ codeunit 1596 "Email Installer"
 
         RetenPolAllowedTables.AddAllowedTable(Database::"Email Outbox", Field.FieldNo(SystemCreatedAt), 7);
         RetenPolAllowedTables.AddAllowedTable(Database::"Sent Email", Field.FieldNo(SystemCreatedAt), 7);
-        RetenPolAllowedTables.AddAllowedTable(Database::"Email Inbox", Field.FieldNo(SystemCreatedAt), 7);
 
         if IsInitialSetup then
             UpgradeTag.SetUpgradeTag(GetEmailTablesAddedToAllowedListUpgradeTag());
     end;
 
+    procedure AddRetentionPolicyEmailInboxAllowedTable(ForceUpdate: Boolean)
+    var
+        Field: Record Field;
+        RetenPolAllowedTables: Codeunit "Reten. Pol. Allowed Tables";
+        UpgradeTag: Codeunit "Upgrade Tag";
+        IsInitialSetup: Boolean;
+    begin
+        IsInitialSetup := not UpgradeTag.HasUpgradeTag(GetEmailInboxAddedToAllowedListUpgradeTag());
+        if not (IsInitialSetup or ForceUpdate) then
+            exit;
+
+        RetenPolAllowedTables.AddAllowedTable(Database::"Email Inbox", Field.FieldNo(SystemCreatedAt), 2);
+
+        if IsInitialSetup then
+            UpgradeTag.SetUpgradeTag(GetEmailInboxAddedToAllowedListUpgradeTag());
+    end;
+
+    procedure CreateRetentionPolicySetup(ForceUpdate: Boolean)
+    var
+        RetentionPolicySetup: Record "Retention Policy Setup";
+        RetentionPolicySetupCU: Codeunit "Retention Policy Setup";
+        UpgradeTag: Codeunit "Upgrade Tag";
+        IsInitialSetup: Boolean;
+    begin
+        IsInitialSetup := not UpgradeTag.HasUpgradeTag(GetEmailInboxPolicyAddedToAllowedListUpgradeTag());
+        if not (IsInitialSetup or ForceUpdate) then
+            exit;
+
+        RetentionPolicySetup.SetRange("Table Id", Database::"Email Inbox");
+        if not RetentionPolicySetup.IsEmpty() then
+            exit;
+
+        RetentionPolicySetup.Validate("Table Id", Database::"Email Inbox");
+        RetentionPolicySetup.Validate("Apply to all records", true);
+        RetentionPolicySetup.Validate("Retention Period", RetentionPolicySetupCU.FindOrCreateRetentionPeriod("Retention Period Enum"::"2 Days"));
+        RetentionPolicySetup.Validate(Enabled, true);
+        RetentionPolicySetup.Insert(true);
```

Comment:
Wouldn't this fail on `ForceUpdate = true` if this is already inserted? It also seems it's also only called with ForceUpdate false. Is this parameter needed?

Reply:
No, it wouldn't fail. There's a check above that the record doesn't exist. It'd exit if it exists.